### PR TITLE
Flat rate whitelist support [SG-280]

### PR DESCRIPTION
**Pull request by jason-c-child**
The intent of these changes are to add a flat-rate whitelist to Names. The scope of changes include

- Add a `updatable-whitelist-flatrate` contract (largely a modified version of the original whitelist code)
- The ability to add either type of whitelist to the `whitelists` array in `name-minter`
- Update discount related code to use both whitelist types
- Update add/remove whitelist functions to support both types

### CODE REVIEW

**Comment by jason-c-child** [comment]
this
needed
so
many
linter
updates
😭

**Review by shanev — COMMENTED** [review]
It looks like this PR includes a version update to CW that changes code in many of the contracts along with the new whitelist. Can they broken into two PRs to make it easier to review? Maybe the CW update first?

**Comment by jason-c-child** [comment]
I've updated this PR to revert the updating of the deprecated functions which is lighting up the linter. I'm not sure why its failing the linter for this while the base branch isn't.

File: contracts/name-minter/src/contract.rs [hunk 1]
```diff
@@ -166,15 +181,33 @@ pub fn execute_mint_and_list(
...
+    });
+
+    let price = validate_payment(
+        name.len(),
+        &info,
+        params.base_price.u128(),
+        discount,
+        whitelist_type.unwrap_or(None),
+    )?;
     charge_fees(&mut res, params.fair_burn_percent, price.amount);
```
[the remark this answers is not in the document]
Comment (jason-c-child) [reply contracts/name-minter/src/contract.rs:211]: Great, I'll cherry pick those and merge them in later today.

File: contracts/name-minter/src/contract.rs [hunk 2]
```diff
@@ -154,10 +165,14 @@ pub fn execute_mint_and_list(
...
+
+    let list = whitelists.iter().find(|whitelist| match whitelist {
+        WhitelistContract::Updatable(updatable) => updatable
+            .includes(&deps.querier, sender.to_string())
+            .unwrap_or(false),
+        WhitelistContract::Flatrate(flatrate) => flatrate
             .includes(&deps.querier, sender.to_string())
-            .unwrap_or(false)
+            .unwrap_or(false),
     });
```
[the remark this answers is not in the document]
Comment (jason-c-child) [reply contracts/name-minter/src/contract.rs]: @MightOfOaks I've merged in the changes from the dYDx-airdrop branch. im curious if we can figure out why this is not matching correctly

**Pull request merged**